iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java

```diff
@@ -1611,7 +1612,68 @@ public SnapshotContext getCurrentSnapshotContext(org.apache.hadoop.hive.ql.metad
     if (current == null) {
       return null;
     }
-    return new SnapshotContext(current.snapshotId());
+    return toSnapshotContext(current);
+  }
+
+  private SnapshotContext toSnapshotContext(Snapshot snapshot) {
```

Comment:
should we put helper private methods at the end of the class?

Reply:
I like them after the method where they called first to easily scroll back to the caller. 
IMHO if the class is too big like this one their location doesn't add any value.